docs/en/routing/basic-usage.md

```diff
@@ -0,0 +1,490 @@
+
+>src/main.tsx
+```tsx
+import renderer from '@dojo/framework/widget-core/vdom';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Registry from '@dojo/framework/widget-core/Registry';
+import { registerRouterInjector } from '@dojo/framework/routing/RouterInjector';
+
+import routes from './routes';
+import App from './App';
+
+const registry = new Registry();
+// creates a router with the routes and registers the router with the registry
+registerRouterInjector(routes, registry);
+
+const r = renderer(() => <App />);
+r.mount({ registry });
+```
+
+Use the `Outlet` widget to control what widgets to render when the path for an outlet id is matched, the example will render `<div>Home</div>` for the `home` outlet.
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={() => <div>Home</div>} />
+			</div>
+		);
+	}
+}
+```
+
+## Path & Query Parameters
+
+Path parameters are placeholders in the routing configuration that will match any value for the segment. The parameters are defined using curly braces, for example: `{param}`.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}',
+		outlet: 'home'
+	}
+];
+```
+
+The parameters values are injected into to matching `Outlets`'s `renderer` property.
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={(matchDetails) => <div>{`Home ${matchDetails.params.page}`}</div>} />
+			</div>
+		);
+	}
+}
+```
+
+Query parameters are also defined using curly braces split from the route path by a `?`. To define more than one query param, use the `&` delimiter.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}?{queryOne}&{queryOne}',
+		outlet: 'home'
+	}
+];
+```
+
+The query params are injected into matching `Outlet` `renderer` properties accessed from `queryParams` property.
+
+## Default Route & Parameters
+
+To add a default route for your application update the routing configuration to include `defaultRoute: true` for the preferred route.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		defaultRoute: true
+	}
+];
+```
+
+If the default route has path or query parameters a map of defaults need to be specified.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}',
+		outlet: 'home',
+		defaultRoute: true,
+		defaultParams: {
+			page: 'about'
+		}
+	}
+];
+```
+
+## Route Configuration
```

Comment:
I don't think we should have a separate section for this should we? It'd be better to tell people about the config as we approach each subject.

Reply:
Extra routing configuration features are gradually introduced as required for each subject, I still think it's valuable to have a section dedicated to the config.

Reply:
What does the extra section give us?

Reply:
It's giving me a table of things that given reading the above is absolutely redundant

Reply:
It gives a summary of the routing configuration, I don't think it's redundant to have an overview section.

Reply:
okay, good for a separate file as already said then. I read this top down as naively as possible and it seemed redundant

Reply:
Cool, I started with everything in the same file - so haven't started to split this out yet. Mostly wanted to get some feed back on the words/examples etc 👍

Reply:
I think these tables add value as there are multiple ways to approach the guide; linear or ad-hoc. Having a complete reference in one place helps the latter usage.

Reply:
Moved to a separate file